### ISSUE TITLE
False positive from `Lint/UselessAssignment` with destructuring assignment

### ISSUE THREAD

**Issue by JF-Lalonde** [rubocop/rubocop#10299]
Attempting to use destructuring assignment with an array triggers `Lint/UselessAssignment` when it shouldn't.

--------

## Expected behavior

The following shouldn't trigger the linter, yet it does:

```
  even, odd = (1..6).partition(&:even?)
```

## Actual behavior

```
Offenses:

E: Lint/UselessAssignment: Useless assignment to variable - even. Use _ or _even as a variable name to indicate that it won't be used. (https://rubystyle.guide#underscore-unused-vars)
      even, odd = (1..6).partition(&:even?)
      ^^^^
E: Lint/UselessAssignment: Useless assignment to variable - odd. Use _ or _odd as a variable name to indicate that it won't be used. (https://rubystyle.guide#underscore-unused-vars)
      even, odd = (1..6).partition(&:even?)
            ^^^
```

## Steps to reproduce the problem

Attempting to assign multiple variables at once to an array erroneously raises the `Lint/UselessAssignment` rule.
This isn't limited to multidimensional arrays, the following also raises a false positive:

```
one, two = [1, 2]
```

## RuboCop version

```
$ bundle exec rubocop -V
1.23.0 (using Parser 3.0.3.1, rubocop-ast 1.13.0, running on ruby 2.7.5 x86_64-darwin20)
  - rubocop-performance 1.12.0
  - rubocop-rails 2.12.4
  - rubocop-rspec 2.6.0
```

**Comment by JF-Lalonde** [comment]
🤦 The linter was triggered because I wasn't referencing the variables after assigning them, mea culpa.
